api/utils/keys/yubikey.go

```diff
@@ -481,7 +539,52 @@ func (y *YubiKey) getPrivateKey(slot piv.Slot) (*PrivateKey, error) {
 		return nil, trace.Wrap(err)
 	}
 
-	return NewPrivateKey(priv, keyPEM)
+	key, err := NewPrivateKey(priv, keyPEM)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return key, nil
+}
+
+// SetPin sets the YubiKey PIV PIN.
+func (y *YubiKey) SetPIN(oldPin, newPin string) error {
+	yk, err := y.open()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer yk.Close()
+
+	err = yk.SetPIN(oldPin, newPin)
+	return trace.Wrap(err)
+}
+
+// checkOrSetPIN prompts the user for PIN and verifies it with the YubiKey.
+// If the user provides the default PIN, they will be prompted to set a
+// non-default PIN and PUK before continuing.
+func (y *YubiKey) checkOrSetPIN(ctx context.Context) error {
+	pin, err := prompt.Password(ctx, os.Stderr, prompt.Stdin(), "Enter your YubiKey PIV PIN [blank to set PIN from default]")
```

Comment:
```suggestion
	pin, err := prompt.Password(ctx, os.Stderr, prompt.Stdin(), "Enter your YubiKey PIV PIN [blank to use default PIN]")
```

Is that correct? I don't totally follow the prompt.

Reply:
Yes that works too. I'll update the RFD to show the prompt flow, any suggestions are welcome. It's a bit of a pain to have the user set 2 PINs but there isn't really an alternative that I can think of.
```bash
> tsh login --user=dev
Enter your YubiKey PIV PIN [blank to use default PIN]:
# \n
The default PIN 123456 is not supported.
Please set a new 6 digit PIN:
Enter your new YubiKey PIV PIN:
Enter your new YubiKey PIV PIN again to confirm:
Enter your YubiKey PIV PUK to reset PIN [blank to use default PUK]
# \n
The default PUK 12345678 is not supported
Please set a new 8 digit PUK:
Enter your new YubiKey PIV PUK:
Enter your new YubiKey PIV PUK again to confirm:
```

Reply:
Your call, I mostly wanted to clarify. FYI I don't usually keep updating RFDs for minor changes like user messages (nor do I go back after a certain time has passed).

Reply:
Yeah, I wasn't sure what to do. Seems like some things are worth adding back to the RFD but probably not this. I'll just add it to the docs next time.